packages/google_maps_flutter/ios/Classes/GoogleMapController.m

```diff
@@ -185,6 +185,23 @@ - (void)onMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
       [_markersController removeMarkerIds:markerIdsToRemove];
     }
     result(nil);
+  } else if ([call.method isEqualToString:@"markers#showInfoWindow"]) {
+    id markerId = call.arguments[@"markerId"];
+    if ([markerId isKindOfClass:[NSString class]]) {
+      [_markersController showMarkerInfoWindow:markerId];
+    }
+    result(nil);
```

Comment:
Here we are returning the future for all cases. Maybe implement an error if markerId is not a String. This is extremely rare, but we might catch errors in the future if dart implementation is modified. Same below:

Something like 

```dart
if (![markerId isKindOfClass:[NSString class]]) {
  result([FlutterError ...]);
  return;
}
result(nil);
```

Reply:
so, should i return an error when markerId is not a NSString and when it's not found on `_markersController`?

Reply:
@giancarlopro sounds good. As for error message, we can say `Invalid markerId` for both cases.